### ISSUE TITLE
Troubles getting email configured/working

### ISSUE THREAD

**Issue by tim-rohrer** [overleaf/toolkit#182]
I've got Docker Mailserver (DMS) running in a container alongside Sharelatex, but I'm unable to get my Overleaf instance to send password reset emails.

## Steps to Reproduce

1. Set variables in `compose.yml` for the `sharelatex` service. Have tried various options. `example.com` is my registered domain, and I've successfully run email tests from it.
```sh
      SHARELATEX_ADMIN_EMAIL: admin@example.com

      # SHARELATEX_LEFT_FOOTER: '[{"text": "Powered by <a href=\"https://www.sharelatex.com\">ShareLaTeX</a> 2016"},{"text": "Another page I want to link to can be found <a href=\"here\">here</a>"} ]'
      # SHARELATEX_RIGHT_FOOTER: '[{"text": "Hello I am on the Right"} ]'

      SHARELATEX_EMAIL_FROM_ADDRESS: "admin@example.com"

      # SHARELATEX_EMAIL_AWS_SES_ACCESS_KEY_ID:
      # SHARELATEX_EMAIL_AWS_SES_SECRET_KEY:

      SHARELATEX_EMAIL_SMTP_HOST: mail
      #SHARELATEX_EMAIL_SMTP_PORT: 465
      SHARELATEX_EMAIL_SMTP_SECURE: true
      SHARELATEX_EMAIL_SMTP_USER: admin@example.com
      SHARELATEX_EMAIL_SMTP_PASS: nopassset
      SHARELATEX_EMAIL_SMTP_TLS_REJECT_UNAUTH: true
      SHARELATEX_EMAIL_SMTP_IGNORE_TLS: false
      SHARELATEX_EMAIL_SMTP_NAME: 'mail.example.com'
      SHARELATEX_EMAIL_SMTP_LOGGER: true
      SHARELATEX_CUSTOM_EMAIL_FOOTER: "This system is run by Tim"
```
2. Proceed to Sharelatex site password reset page, enter email address and click Request password reset.

## Expected Behaviour

I should get a password reset email.

## Observed Behaviour

<img width="374" alt="image" src="https://github.com/overleaf/toolkit/assets/7756865/18affda8-674b-4868-a02e-324f4164c998">

## Context

As I don't remember my password for the user in my Overleaf instance, I'm unable to use the app or get access to my docs.

## Technical Info

Output to `web.log` when password reset requested:
```json
{
  "name": "web",
  "hostname": "3000ca4df1a3",
  "pid": 135,
  "level": 50,
  "err": {
    "message": "error sending message",
    "name": "OError",
    "stack": "OError: error sending message\n    at Object.sendEmail (/overleaf/services/web/app/src/Features/Email/EmailSender.js:100:11)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async Object.sendEmail (/overleaf/services/web/app/src/Features/Email/EmailHandler.js:23:3)\nTaggedError: failed to generate and email password reset token\n    at /overleaf/services/web/app/src/Features/PasswordReset/PasswordResetController.js:107:16\n    at Object.<anonymous> (/overleaf/services/web/app/src/Features/PasswordReset/PasswordResetHandler.js:32:18)\n    at __node_internal_ (node:util:281:10)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)\ncaused by:\n    Error: connect ECONNREFUSED 127.0.0.1:587\n        at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1187:16)",
    "info": { "email": "tim@myemail.com" }
  },
  "url": "/user/password/reset",
  "method": "POST",
  "user": null,
  "msg": "error passed to top level next middleware",
  "time": "2023-08-05T05:22:52.108Z",
  "v": 0
}
```

## Analysis

I've tried many settings for the SMTP variables but every time the log reports: `Error: connect ECONNREFUSED 127.0.0.1:587` Even if I change ports to which it should connect. 

Why does Sharelatex always try to connect to the localhost (I'm assuming the host on port 587)? The DMS service is named `mail` and it is set as the `SHARELATEX_EMAIL_SMTP_HOST`. I've also tried setting `SHARELATEX_EMAIL_SMTP_PORT` to ports 465 and 25 with the same result. 

My apps are in one `compose.yml` file, and they are all on the same bridge network.

I was able to `docker exec -it sharelatex bash` and `telnet mail 25` to successfully send a test email message. From this, I'm concluding the issue I'm facing is a problem with my variables shown above, or some sort of bug with Sharelatex/Overleaf.

**Comment by tim-rohrer** [comment]
It occurred to me that I was likely using an old image. I tried upgrading to 4, but that didn't work with my current MongoDB, and I don't want to update that until I can get into Overleaf and back up my files.

I reverted to the latest 3 and am now getting a different error: `Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames: Host: mail. is not in the cert's altnames`

I consider this progress and will continue to try different configurations.
